### DISCUSSION TITLE
Flattening `lib`, such as `lib/git` into `git`

### DISCUSSION THREAD

**Discussion by ten3roberts** [NeogitOrg/neogit#598] [Ideas]
The current structure is very deep, and as such leads to long `require` statements, as well as possibly hurting discoverability as it is more difficult to get an overview of the deeply nested folders.

What do you think about moving some of the "larger" subdirectories inside lib out one level. For example the `git` subdirectory?

**Comment by CKolkey** [comment]
Hmm. I agree with the problem, but I'm not convinced about the solution. I _do_ agree that the structure isn't strictly adhered to - what is a `lib` and what isn't? I think the `git` stuff could be considered a library, as it's the wrapper for the git cli.

Another way to solve the discoverability issue would be to provide an API breakdown in the `CONTRIBUTORS.md`, which would solve another issue too :)

**Reply by ten3roberts** [reply]
Like this one https://github.com/NeogitOrg/neogit/pull/599 :blush:

**Reply by CKolkey** [reply]
Hell yea :D
